Python/ast_unparse.c

```diff
@@ -61,13 +64,30 @@ append_charp(_PyUnicodeWriter *writer, const char *charp)
 static int
 append_repr(_PyUnicodeWriter *writer, PyObject *obj)
 {
-    int ret;
-    PyObject *repr;
-    repr = PyObject_Repr(obj);
+    if (PyFloat_CheckExact(obj) && Py_IS_INFINITY(PyFloat_AS_DOUBLE(obj))) {
+        return _PyUnicodeWriter_WriteStr(writer, _str_replace_inf);
```

Comment:
Wait, it should be _str_inf, no?

Reply:
Hum, _str_replace_inf looks correct. But test_future fails:

```
======================================================================
FAIL: test_infinity_numbers (test.test_future.AnnotationsFutureTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vstinner/python/master/Lib/test/test_future.py", line 303, in test_infinity_numbers
    self.assertAnnotationEqual("-1e1000", expected=f"-{inf}")
  File "/home/vstinner/python/master/Lib/test/test_future.py", line 142, in assertAnnotationEqual
    self.assertEqual(actual, expected)
AssertionError: '1e309' != '-1e309'
- 1e309
+ -1e309
? +
```

I suggest you to alway run tests locally before pushing :-)

```
./python -m test -j0 test_unparse test_future 
```

Reply:
Test failure looks correct because we aren't handling what if it is a -inf, I guess we should revert it back to the first form where we render over repr

Reply:
Please don't listen to me when I give you wrong advices :-D I forgot about -inf!